disperser/controller/metadata/batch_metadata_manager.go

```diff
@@ -150,7 +124,7 @@ func (m *batchMetadataManager) updateMetadata() error {
 		return nil
 	}
```

Comment:
Question about this `if` block:

Why are you checking `referenceBlockNumber == previousMetadata.referenceBlockNumber` instead of `referenceBlockNumber <= previousMetadata.referenceBlockNumber`?